src/main/java/com/kryptnostic/rhizome/hazelcast/serializers/SetStreamSerializers.java

```diff
@@ -47,15 +47,32 @@
     }
 
     public static void fastUUIDSetSerialize( ObjectDataOutput out, Iterable<UUID> object ) throws IOException {
-        long[] least = new long[ Iterables.size( object ) ];
-        long[] most = new long[ Iterables.size( object ) ];
+        int length = Iterables.size( object );
+        long[] least = new long[ length ];
+        long[] most = new long[ length ];
         int i = 0;
         for ( UUID uuid : object ) {
             least[ i ] = uuid.getLeastSignificantBits();
             most[ i ] = uuid.getMostSignificantBits();
             i++;
         }
-        out.writeInt( i );
+        out.writeInt( length );
```

Comment:
I'm not sure we need this because writeLongArray records its own length.

Reply:
Oh yeah, that's a really good point actually

Reply:
You don't need out.writeInt(length) either.

Reply:
in.readLongArray(least) will automatically read the correct length array.

Reply:
in.readLongArray(least) will automatically read the correct length array.